In [1]:
import pandas as pd
df = pd.read_csv("./clear_data.csv")
df

Invoice_ID Branch       City  ... Gross_income Cost_price Unit_cost
0     765-26-6951      A     Yangon  ...        68.62     388.82     64.80
1     530-90-9855      A     Yangon  ...        71.96     327.80     40.98
2     891-01-7034      B   Mandalay  ...        70.60     400.07     66.68
3     493-65-6248      C  Naypyitaw  ...        58.24     330.05     33.01
4     556-97-7101      C  Naypyitaw  ...        19.91     112.85     56.43
...           ...    ...        ...  ...          ...        ...       ...
1164  291-55-6563      A     Yangon  ...        39.03     177.82     29.64
1165  361-85-2571      A     Yangon  ...        70.47     399.30     79.86
1166  364-34-2972      C  Naypyitaw  ...        45.75     259.23     86.41
1167  115-38-7382      C  Naypyitaw  ...         9.28      27.84      3.48
1168  115-38-7382      C  Naypyitaw  ...         9.94      29.81      3.73

[1169 rows x 16 columns]

**1. коэффициенты корреляции между продажами товарных групп**

In [3]:
# Сгруппируем продажи по товарным группам с помощью сводной таблицы
df_pt = pd.pivot_table(df[["Date", "Product_line", "Total"]], values=["Total"], index=["Date"], columns="Product_line")
df_pt

Total  ...                  
Product_line                  Clothes and accessories  ... Sports and travel
Date                                                   ...                  
2019-01-01 00:00:00.000000000                267.3800  ...        285.847500
2019-01-02 00:00:00.000000000                123.3400  ...        197.263333
2019-01-03 00:00:00.000000000                     NaN  ...         98.050000
2019-01-04 00:00:00.000000000                128.1900  ...        146.220000
2019-01-05 00:00:00.000000000                 71.6875  ...               NaN
...                                               ...  ...               ...
2019-03-26 00:00:00.000000000                 22.5200  ...         41.080000
2019-03-27 00:00:00.000000000                     NaN  ...        588.360000
2019-03-28 00:00:00.000000000                 87.2200  ...               NaN
2019-03-29 00:00:00.000000000                     NaN  ...               NaN
2019-03-30 00:00:00.000000000                     NaN  ...        382.730000

[90 rows x 7 columns]

In [12]:
# Построим матрицу корреляции между товарными категориями:
df_pt.corr()

Total  ...                  
Product_line                  Clothes and accessories  ... Sports and travel
      Product_line                                     ...                  
Total Clothes and accessories                1.000000  ...         -0.084375
      Electronic accessories                 0.020188  ...         -0.092085
      Fashion accessories                    0.674053  ...          0.026315
      Food and beverages                     0.300518  ...         -0.147570
      Health and beauty                      0.006577  ...         -0.094024
      Home and lifestyle                     0.037058  ...         -0.062088
      Sports and travel                     -0.084375  ...          1.000000

[7 rows x 7 columns]

Из матрицы корреляции видим, что товарные группы мало коррелируют между собой, самая высокая корреляция у товарных групп "Clothes and accessories" и "Fashion accessories", составляет **0.674**

**2. коэффициент корреляции между продажами Fashion accessories и Clothes and accessories для мужчин и для женщин**

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1169 entries, 0 to 1168
Data columns (total 16 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Invoice_ID     1169 non-null   object 
 1   Branch         1169 non-null   object 
 2   City           1169 non-null   object 
 3   Customer_type  1169 non-null   object 
 4   Gender         1169 non-null   object 
 5   Product_line   1169 non-null   object 
 6   Unit_price     1169 non-null   float64
 7   Quantity       1169 non-null   int64  
 8   Total          1169 non-null   float64
 9   Date           1169 non-null   object 
 10  Time           1169 non-null   object 
 11  Payment        1169 non-null   object 
 12  Payments_fee   1169 non-null   float64
 13  Gross_income   1169 non-null   float64
 14  Cost_price     1169 non-null   float64
 15  Unit_cost      1169 non-null   float64
dtypes: float64(6), int64(1), object(9)
memory usage: 146.3+ KB


**Мужчины**

In [18]:
# отфильтруем продажи мужчинам по интересующим нас товарным группам
df_men = df[(df["Gender"] == "Male")&((df["Product_line"] == "Fashion accessories")|(df["Product_line"] == "Clothes and accessories"))]

# ссумируем продажи мужчинам по датам
df_men_pt = pd.pivot_table(df_men, values=["Total"], index=["Date"], columns="Product_line")

# посмотрим корреляцию между продажами товарных групп:
df_men_pt.corr()

Total                    
Product_line                  Clothes and accessories Fashion accessories
      Product_line                                                       
Total Clothes and accessories                1.000000            0.030221
      Fashion accessories                    0.030221            1.000000

Из таблицы видно, что корреляция между продажами "Fashion accessories" и "Clothes and accessories" для мужчин низкая

**Женщины**

In [19]:
# отфильтруем продажи женщинам по интересующим нас товарным группам
df_women = df[(df["Gender"] == "Female")&((df["Product_line"] == "Fashion accessories")|(df["Product_line"] == "Clothes and accessories"))]

# ссумируем продажи мужчинам по датам
df_women_pt = pd.pivot_table(df_women, values=["Total"], index=["Date"], columns="Product_line")

# посмотрим корреляцию между продажами товарных групп:
df_women_pt.corr()

Total                    
Product_line                  Clothes and accessories Fashion accessories
      Product_line                                                       
Total Clothes and accessories                1.000000            0.996301
      Fashion accessories                    0.996301            1.000000

Из таблицы видно, что корреляция между продажами "Fashion accessories" и "Clothes and accessories" для женщин очень высокая. Можно уверенно сказать, что продажи этих двух товарных групп у женщин тесно взаимосвязана